web/src/settings_ui.ts

```diff
@@ -34,7 +34,7 @@ export const strings = {
 export function do_settings_change(
     request_method: AjaxRequestHandler,
     url: string,
-    data: Parameters<AjaxRequestHandler>[0]["data"],
+    data: Omit<Parameters<AjaxRequestHandler>[0]["data"], "undefined">,
```

Comment:
Why does this need an `omit`?

Reply:
Updated the PR with an explanation for the `omit` .